app/routes.py

```diff
@@ -1,2 +1,446 @@
-from flask import Blueprint
+from flask import Blueprint, make_response, jsonify, request
+from app import db
+from app.models.video import Video
+from app.models.customer import Customer
+from app.models.rental import Rental
+from datetime import datetime
+import re
 
+
+customers_bp = Blueprint("customers", __name__, url_prefix="/customers")
+videos_bp = Blueprint("videos", __name__, url_prefix="/videos")
+rentals_bp = Blueprint("rentals", __name__, url_prefix="/rentals")
+
+# ----------------- helper functions ------------------------
+
+
+def is_number(input):
+    condition = isinstance(input, int) or isinstance(input, float)
+    return True if condition else False
+
+
+def is_valid_phone(input):
+    """Validates the phone number pattern (908) 949-6758"""
+    pattern = re.compile(r'\(\d{3}\) \d{3}-\d{4}')
+    return re.match(pattern, input) is not None
+
+
+# -------------- CRUD for /customers ------------------------
+
+
+@customers_bp.route("", methods=["GET"])
+def get_all_customers():
+    """Lists all existing customers and details about each customer."""
+    customers = Customer.query.all()
+    response = [customer.to_json() for customer in customers]
+    return jsonify(response), 200
+
+
+@customers_bp.route("/<customer_id>", methods=["GET"])
+def get_customer_by_id(customer_id):
+    """Gives back details about specific customer."""
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response({
+            "errors": [
+                "Not Found",
+                "Customer does not exist"
+            ]
+        }, 404)
+    return make_response(customer.to_json(), 200)
+
+
+@customers_bp.route("", methods=["POST"])
+def add_customer():
+    """Creates a new customer with the given Request Body Parameters."""
+    request_body = request.get_json()
+
+    if not (
+            "name" in request_body and
+            "postal_code" in request_body and
+            "phone" in request_body
+            # and
+            # is_number(request_body["postal_code"]) and
+            # is_valid_phone(request_body["phone"])
+    ):
+        return make_response({
+            "errors": [
+                "Bad Request",
+                "'name' is required",
+                "'postal_code' is required and should be a number",
+                "'phone' is required and should be (XXX) XXX-XXXX"
+            ]
+        }, 400)
+
+    new_customer = Customer()
+    new_customer = new_customer.from_json(request_body)
+    db.session.add(new_customer)
+    db.session.commit()
+
+    retrieve_customer = Customer.query.get(new_customer.customer_id)
+    return make_response(retrieve_customer.to_json(), 201)
```

Comment:
Double check, but I think you can save yourself another call to your database by passing `new_customer` into the response instead. It should already come with the new id.

```suggestion
    return make_response(new_customer.to_json(), 201)
```